azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/ConfigMojo.java

```diff
@@ -372,10 +417,177 @@ private String getDefaultValue(String defaultValue, String fallBack, String patt
     }
 
     private boolean isJarProject() {
-        return getProject().getPackaging().equalsIgnoreCase("jar");
+        return Utils.isJarPackagingProject(project.getPackaging());
     }
 
     private WebAppConfiguration getWebAppConfigurationWithoutValidation() throws AzureExecutionException {
         return new V2NoValidationConfigurationParser(this, new V2ConfigurationValidator(this)).getWebAppConfiguration();
     }
+
+    private WebAppConfiguration chooseExistingWebappForConfiguration()
+            throws AzureExecutionException, AzureAuthFailureException {
```

Comment:
this method seems too long. is it possible to slit it into steps(well-named methods).